conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -0,0 +1,346 @@
+                exception.initCause(e);
+                return exception;
+            }
+
+            Optional<String> contentType = Optional.ofNullable(
+                            response.headers().get(HttpHeaders.CONTENT_TYPE))
+                    .map(values -> Iterables.getFirst(values, null));
+            if (contentType.isPresent()
+                    && contentType.get().toLowerCase(Locale.ENGLISH).startsWith("application/json")) {
+                try {
+                    SerializableError serializableError = MAPPER.readValue(body, SerializableError.class);
+                    return new RemoteException(serializableError, response.status());
+                } catch (Exception e) {
+                    return new UnknownRemoteException(response.status(), body);
+                }
+            }
+
+            return new UnknownRemoteException(response.status(), body);
+        }
+
+        private static String toString(InputStream body) throws IOException {
+            try (Reader reader = new InputStreamReader(body, StandardCharsets.UTF_8)) {
+                return CharStreams.toString(reader);
+            }
+        }
+
+        @Override
+        public Exception decode(String methodKey, feign.Response response) {
+            if (isError(response)) {
+                return parse(response);
+            }
+            return defaultDecoder.decode(methodKey, response);
+        }
+    }
+
+    private final class FeignRequestEndpoint implements Endpoint {
+        private final feign.Request request;
+        private final String endpoint;
+
+        FeignRequestEndpoint(feign.Request request) {
+            this.request = request;
+            endpoint = getFirstHeader(request, PATH_TEMPLATE).orElse("feign");
+        }
+
+        @Override
+        public void renderPath(Map<String, String> _params, UrlBuilder url) {
+            String target = request.url();
+            Preconditions.checkState(
+                    target.startsWith(baseUrl),
+                    "Request URL must start with base url",
+                    UnsafeArg.of("baseUrl", baseUrl));
+            String trailing = target.substring(baseUrl.length());
+            int queryParamsStart = trailing.indexOf('?');
+            String queryPortion = queryParamsStart == -1 ? trailing : trailing.substring(0, queryParamsStart);
+            for (String pathSegment : pathSplitter.split(queryPortion)) {
+                url.pathSegment(urlDecode(pathSegment));
+            }
+            if (queryParamsStart != -1) {
+                String querySegments = trailing.substring(queryParamsStart + 1);
+                querySplitter
+                        .split(querySegments)
+                        .forEach((name, value) -> url.queryParam(urlDecode(name), urlDecode(value)));
+            }
+        }
+
+        @Override
+        public HttpMethod httpMethod() {
+            return HttpMethod.valueOf(request.method().toUpperCase(Locale.ENGLISH));
```

Comment:
could store this in a field to avoid recomputing the same thing every time we call this method?